### PR TITLE
Generate votes only if the proposed bank is not censoring the validator

### LINKED ISSUE #6727: Validators may continue voting even if the network is censoring the validator
#### Problem
Validator may continue voting even if the network is censoring the validator.

#### Proposed Solution

Validator should stop voting if the bank is more than the lockout threshold behind the validator.

Validator can let some votes expire and retry.

### PR DESCRIPTION
**Pull request by aeyakovenko**
#### Problem

Validators may continue voting even though the network is dropping the votes.

#### Summary of Changes

If the delta between expected vote state and current vote state > THRESHOLD, consensus stops voting and lets some votes expire.

Fixes #6727

### CODE REVIEW

File: core/src/replay_stage.rs [hunk 1]
```diff
@@ -493,56 +493,66 @@ impl ReplayStage {
...
+                    .write()
+                    .unwrap()
+                    .push_vote(tower_index, vote_tx);
+            }
+        } else {
+            warn!(
+                "Vote dropped for bank {}, bank censorship failure!!!",
+                bank.slot()
+            );
+            inc_new_counter_info!("replay_stage-votable_bank_censorship_failure", 1);
```
Comment (aeyakovenko) [inline core/src/replay_stage.rs]: @sagar-solana @carllin 

how do I plum the counter into the dashboard these days?

Comment (carllin) [reply core/src/replay_stage.rs]: hehe it's kind of a PIA. Usually you create the dashboard you want in Grafana, then copy the JSON model into `testnet-monitor.json`

File: core/src/consensus.rs [hunk 2]
```diff
@@ -829,12 +844,34 @@ mod test {
...
+    fn test_new_vote_threshold() {
+        let mut local = VoteState::default();
+        //network is censoring
+        let vote = Vote {
+            slots: vec![0, 1, 2, 3],
+            hash: Hash::default(),
+        };
+        local.process_vote_unchecked(&vote);
+        //vote 4 should be at threshold 5
+        assert_eq!(Tower::new_vote(4, &local, 4, Hash::default(), None), None);
```
Comment (carllin) [inline core/src/consensus.rs]: Maybe one more test case where `last_bank_slot` is not None to exercise that code path?

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Comment by carllin** [comment]
@aeyakovenko, the local cluster catchup tests won't pass unless the validators playing catch up set roots (issue we discussed yesterday).

**Comment by aeyakovenko** [comment]
@carllin want to take this over?

**Comment by aeyakovenko** [comment]
@mvines i don’t think we need this for 0.20

**Comment by mvines** [comment]
@carllin - can you either rebase and help get this PR moving again, or just close it if obsolete.

**Comment by garious** [comment]
@carllin, what's going on with this PR?

**Comment by carllin** [comment]
On hold pending design of how to update root without voting so that validators catching up from a snapshot/reboot can update roots even while votes aren't landing

**Comment by mvines** [comment]
Ok let's close this PR for now.  Stale bot will again anyway, and there's a tracking issue so we won't forget about it

**Pull request closed without merging**